quick patch for PackPadded removal to propagate the correct size. (#9042)

anderspapitto opened the pull request:
Summary:
The underlying reason why this is even an issue is that the conversion
into and out of the 'fictional' onnx operators is done in an unhygenic
order. This doesn't address that, but it does fix the one observable
case where this produces an incorrect result, and unblocks some other
work being done.
Closes https://github.com/pytorch/pytorch/pull/9042

Differential Revision: D8700364

ezyang commented on torch/csrc/jit/passes/onnx/peephole.cpp:
```diff
@@ -259,6 +259,16 @@ void pushPackingPastRnn(Block *b) {
     newPackPadded->addInput(next->outputs()[0]);
     newPackPadded->addInput(n->inputs()[1]);
 
+    TensorType* oldType = rnn->inputs()[0]->type()->cast<TensorType>();
```
Temporary hack code should have a comment explaining what's going on

ezyang commented:
CC @colesbury

anderspapitto committed:
quick patch for PackPadded removal to propagate the correct size. (#9189)

Summary:
Closes https://github.com/pytorch/pytorch/pull/9189

The underlying reason why this is even an issue is that the conversion
into and out of the 'fictional' onnx operators is done in an unhygenic
order. This doesn't address that, but it does fix the one observable
case where this produces an incorrect result, and unblocks some other
work being done.
Closes https://github.com/pytorch/pytorch/pull/9042

Differential Revision: D8700364

fbshipit-source-id: 55ed6c01584f3c8ad401fba418fabfa66361f9ca

ezyang commented:
Let's merge it.

anderspapitto commented:
import/export tooling is confused on this one, will open a new PR
